.github/workflows/build-and-test.yml

```diff
@@ -135,28 +134,25 @@ jobs:
         tags: ${{ env.TEST_IMAGE_TAG }}
         target: all-tools
         cache-from: type=registry,ref=${{ env.REGISTRY }}/${{ github.repository_owner }}/ort:cache
-    - name: Setup Gradle
+    - name: Set step output
+      id: final
+      run: echo "test_image_tag=$TEST_IMAGE_TAG" >> "$GITHUB_OUTPUT"
+  funTest-docker:
+    needs: funTest-docker-build
+    runs-on: ubuntu-22.04
+    container:
+      image: ${{ needs.funTest-docker-build.outputs.test_image_tag }}
+      options: --user 1001
```

Comment:
> I think publishing PR builds next to the official builds in the registry is problematic

Why? Wouldn't it be ok if we tagged them uniquely and removed them after the PR build again?

Reply:
I guess I could live with that, even though I don't like those builds to appear when someone is looking at the packages in the GitHub UI. However, making sure that the image gets deleted even if some other steps fail might require implementing a custom action, I think composite actions don't support cleanup steps.
Another question is that of performance, of course the container job syntax is nicer than what we currently have, but it would have to be tested how uploading the image to the registry and then downloading it again compares to the current implementation.

Reply:
> I think composite actions don't support cleanup steps.

Wouldn't something like `if: ${{ always() }}` work, like we have it in `static-analysis.yml`?

> Another question is that of performance, of course the container job syntax is nicer than what we currently have

I agree that performance matters, but this is not only about syntactic sugar. It also implicitly fixes the issue of restoring the unified diff for test failures, as environment variables get properly forwarded. Solving that problem led me to first look into why it worked before (with using `container`).

Reply:
> > I think composite actions don't support cleanup steps.
> 
> Wouldn't something like `if: ${{ always() }}` work, like we have it in `static-analysis.yml`?

If that guarantees that the step is always executed that should be fine.

> > Another question is that of performance, of course the container job syntax is nicer than what we currently have
> 
> I agree that performance matters, but this is not only about syntactic sugar. It also implicitly fixes the issue of restoring the unified diff for test failures, as environment variables get properly forwarded. Solving that problem led me to first look into why it worked before (with using `container`).

Forwarding the kotest variable to Docker would also be trivial, so I would give performance the priority here.